### PR TITLE
feat(discover): Add a toggle to hide prebuilt queries

### PR DESCRIPTION
**Pull request by wmak**
- This adds a toggle that lets a user hide prebuilt queries
- This is toggle is only enabled when there's at least 1 other query
  already so that they don't end up with an empty discover
image

### CODE REVIEW

File: src/sentry/static/sentry/app/views/eventsV2/landing.tsx [hunk 1]
```diff
@@ -373,7 +408,7 @@ const StyledSearchBar = styled(SearchBar)`
 const StyledActions = styled('div')`
   display: grid;
   grid-gap: ${space(2)};
-  grid-template-columns: auto min-content;
+  grid-template-columns: auto max-content min-content;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/landing.tsx:411]: Does this need a different layout in mobile/tablet viewports?

Comment (wmak) [reply src/sentry/static/sentry/app/views/eventsV2/landing.tsx:411]: I made these stack instead on smaller viewports:
image

**Review by dashed — APPROVED** [review]

File: src/sentry/static/sentry/app/views/eventsV2/landing.tsx [hunk 2]
```diff
@@ -258,6 +267,15 @@ class DiscoverLanding extends AsyncComponent<Props, State> {
           placeholder={t('Search saved queries')}
           onSearch={this.handleSearchQuery}
         />
+        <PrebuiltSwitch>
+          <SwitchLabel>Show Prebuilt</SwitchLabel>
+          <Switch
+            isActive={renderPrebuilt}
+            isDisabled={renderPrebuilt && savedQueries.length === 0}
+            size="lg"
+            toggle={() => this.togglePrebuilt()}
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/views/eventsV2/landing.tsx:276]:
```suggestion
            toggle={this.togglePrebuilt}
```

`togglePrebuilt` is already using an arrow function so `this` will be correctly bound. Using another arrow function here can cause more re-renders than necessary.

File: src/sentry/static/sentry/app/views/eventsV2/landing.tsx [hunk 3]
```diff
@@ -274,6 +292,14 @@ class DiscoverLanding extends AsyncComponent<Props, State> {
     );
   }
 
+  togglePrebuilt = () => {
+    const {renderPrebuilt} = this.state;
+
+    setRenderPrebuilt(!renderPrebuilt);
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/views/eventsV2/landing.tsx:298]: Maybe this should be in the callback of `setState` once react finishes updating the state?

**Pull request merged**